packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -1191,6 +1206,39 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
         event.preventDefault();
     }
 
+    /**
+     * Handle the single-click mouse event on the expansion toggle.
+     * @param node the tree node if available.
+     * @param event the mouse single-click event.
+     */
+    protected handleExpansionToggleClickEvent(node: TreeNode | undefined, event: React.MouseEvent<HTMLElement>): void {
```

Comment:
We've backed ourselves into corner a bit on naming, since `toggle` should be `handleExpansionToggleClick` and `doToggle` should be `doHandleExpansionToggleClick`. I'd prefer not to add to the confusion by adding methods by those names that aren't actually click handlers. Once the redundant condition is removed, I don't think there's enough here to justify the method, so just move this up to `doToggle`.